src/targets/Logary.Targets.Zipkin/Targets_Zipkin.fs

```diff
@@ -0,0 +1,141 @@
+﻿namespace Logary.Targets
+
+// https://github.com/racker/restkin
+// https://github.com/racker/tryfer
+/// A target for Zipkin.
+module Zipkin = 
+    open System
+    open System.IO
+    open System.Net
+    open System.Runtime.CompilerServices
+    open Hopac
+    open Hopac.Infixes
+    open Logary
+    open Logary.Internals
+    open Logary.Target
+    open Logary.Formatting
+    open Zipkin
+    open Microsoft.FSharp.Reflection
+        
+    module internal Option = 
+        let orDefault a = Option.fold (fun s t -> t) a
+    
+    open Option
+    
+    /// Configuration for Zipkin target.
+    type ZipkinConf = 
+        { baseUri : Uri
+          collectorFactory : Uri -> ISpanCollector }
+        
+        /// Create a new configuration with a base uri and an optional
+        /// rest client factory.
+        /// Default port: 9411
+        /// Default host: localhost
+        static member Create(?baseUri, ?fac, ?traceReq, ?formatter) = 
+            { baseUri = defaultArg baseUri (Uri "http://localhost:9411")
+              collectorFactory = defaultArg fac (fun uri -> upcast HttpCollector uri) }
+        
+        static member Default = ZipkinConf.Create()
+    
+    type private ZipkinState = 
+        { activeSpans : Map<uint64, Span>
```

Comment:
It's how I would do it, but I would use `Scheduler` to schedule removal of the span after a timeout. You could also use the Cache classes which support timed removal.